ext/background/background.js

```diff
@@ -171,16 +172,42 @@ async function attemptRecordInteraction(tab, session, force) {
     return Promise.resolve();
   }
   setChromeActivity(a, tab);
+
+  const merged = await mergeAttachment(session, a);
   return recordInteraction(
     {
       interaction_type: (_) => 'VIEWED',
       idempotency_key: `${moment().startOf('day')}::${tab.title}`,
-      attachment: a,
+      attachment: merged,
     },
     authorize(session)
   );
 }
 
+// Queries Range for existing attachments and merges the attachments based on a
+// provider's configured behavior. The Range backend will overwrite all fields
+// sent to it, but leave the rest alone.
+async function mergeAttachment(session, attachment) {
+  const dedupe = getProviderDedupe(attachment.provider);
+  if (dedupe == MERGE_BEHAVIOR.REPLACE_EXISTING) return attachment;
+
+  // This is the default, KEEP_EXISTING case
+  const activity = await listActivity(attachment.provider, authorize(session));
```

Comment:
This works for now, but will miss attachments that aren't in recent activity and isn't as efficient as it could be.

It seems like we might want a new a new binding for `ReadAttachment` that takes the source ID, provider, and org ID instead of the attachment ID. `ReadAttachment` already has the logic for checking whether the given user should have access to the given attachment or not. (Or perhaps it's a new endpoint name like `FindAttachment`...)

Reply:
Agreed, this would simplify things a lot! Added to the backlog.